src/js/components/TextArea/TextArea.js

```diff
@@ -1,28 +1,25 @@
-import React, { Component } from 'react';
+import React from 'react';
 import { compose } from 'recompose';
 
 import { Keyboard } from '../Keyboard';
 import { withFocus, withForwardRef } from '../hocs';
 
 import { StyledTextArea } from './StyledTextArea';
 
-class TextArea extends Component {
-  onEsc = event => {
+const TextArea = ({ fill, forwardRef, ...rest }) => {
+  const onEsc = event => {
     // we have to stop both synthetic events and native events
     // drop and layer should not close by pressing esc on this input
     event.stopPropagation();
     event.nativeEvent.stopImmediatePropagation();
   };
 
-  render() {
-    const { fill, forwardRef, ...rest } = this.props;
-    return (
-      <Keyboard onEsc={this.onEsc}>
-        <StyledTextArea ref={forwardRef} fillArg={fill} {...rest} />
-      </Keyboard>
-    );
-  }
-}
+  return (
+    <Keyboard onEsc={onEsc}>
+      <StyledTextArea ref={forwardRef} fillArg={fill} {...rest} />
```

Comment:
I'm wondering if we still need the forward ref here since it is a functional component and not a class. thoughts?

Reply:
probably yes for backwards compatibility.